### PR TITLE
Update on Credential secretname change #155

### LINKED ISSUE #155: React on Identities (endpoint auth) changes
Relates to #81 (subset)

### PR DESCRIPTION
**Pull request by rigazilla**
This PR enables rolling update (restart) if identities (security.endpointSecret) changes
Partially fixes #155 : doesn't detect secret content change atm

Replaces #109

### CODE REVIEW

**Comment by rigazilla** [comment]
@galderz added auth test with credentials change

**Comment by galderz** [comment]
@rigazilla I don't think there's a need to add yet another test. You could just expand `TestExternalServiceWithAuth` so that at the end it tries to modify the auth config...etc. Each test creates a new Infinispan operator instance (for the time being), so the quicker it runs the better.

At some point it'd be nice to have a single operator instance start up before all tests, and then run each test where you scale up to 2 nodes, add security...etc. I think this would both save execution time and would test both the logic and the capabilities of the operator to handle changes.

File: test/e2e/main_test.go [hunk 1]
```diff
@@ -224,6 +226,101 @@ func TestExternalServiceWithAuth(t *testing.T) {
...
+	er := kubernetes.Kubernetes.Client.Update(context.TODO(), &spec)
+	if er != nil {
+		panic(fmt.Errorf(err.Error()))
+	}
+	kubernetes.WaitForPods("app=infinispan-pod", 1, SinglePodTimeout, Namespace)
+
+	// Test authentication
+	routeName := fmt.Sprintf("%s-external", name)
+	client := &http.Client{}
+	hostAddr := kubernetes.WaitForExternalService(routeName, RouteTimeout, client, Namespace)
```
Comment (galderz) [inline test/e2e/main_test.go]: CI failed here with Minikube. Can you have a look?

Comment (galderz) [reply test/e2e/main_test.go]: Run: https://travis-ci.org/infinispan/infinispan-operator/jobs/588876584

Comment (galderz) [reply test/e2e/main_test.go]: To be precise, it got stuck within there.

**Comment by rigazilla** [comment]
> @rigazilla I don't think there's a need to add yet another test. You could just expand `TestExternalServiceWithAuth` so that at the end it tries to modify the auth config...etc. Each test creates a new Infinispan operator instance (for the time being), so the quicker it runs the better.
> 
> At some point it'd be nice to have a single operator instance start up before all tests, and then run each test where you scale up to 2 nodes, add security...etc. I think this would both save execution time and would test both the logic and the capabilities of the operator to handle changes.

ok done @galderz 
Tried on local minikube and test are fine.
Let's see the new run on ci... tomorrow :)

**Comment by rigazilla** [comment]
Minikube test is still failing on CI. Test seems to run until the end, so it must be something in the clean up.
Looking into

**Comment by rigazilla** [comment]
@galderz reset timeout to default and squashed history

**Pull request merged**